helix-view/src/theme/theme_keys.rs

```diff
@@ -0,0 +1,533 @@
+
+    // Keywords
+    "keyword"                      is Keyword                   -> []                                                : "Default style for keywords",
+    "keyword.control"              is KeywordControl            -> [Keyword]                                         : "Control keywords",
+    "keyword.control.conditional"  is KeywordControlConditional -> [KeywordControl, Keyword]                         : "Conditional control keywords, e.g. `if`, `else`, etc.",
+    "keyword.control.exception"    is KeywordControlException   -> [KeywordControl, Keyword]                         : "Exception keywords, e.g. `try`, `raise`, etc.",
+    "keyword.control.import"       is KeywordControlImport      -> [KeywordControl, Keyword]                         : "Import/Include keywords, e.g. `use`, `include`, etc.",
+    "keyword.control.repeat"       is KeywordControlRepeat      -> [KeywordControl, Keyword]                         : "Loop keywords, e.g. `for`, `while`, etc.",
+    "keyword.control.return"       is KeywordControlReturn      -> [KeywordControl, Keyword]                         : "Return keywords, e.g. `return`, `yield`, etc.",
+    "keyword.directive"            is KeywordDirective          -> [Keyword]                                         : "Directives, e.g. `#include`, `comptime`, etc.",
+    "keyword.function"             is KeywordFunction           -> [Keyword]                                         : "Function keywords, e.g. `fn`, `function`, etc.",
+    "keyword.local"                is KeywordLocal              -> [Keyword]                                         : "Local keywords",
+    "keyword.local.import"         is KeywordLocalImport        -> [KeywordLocal, Keyword]                           : "Local imports keywords",
+    "keyword.operator"             is KeywordOperator           -> [Keyword]                                         : "Operator keywords, e.g. `not`, `or`, etc.",
+    "keyword.operator.assignment"  is KeywordOperatorAssignment -> [KeywordOperator, Keyword]                        : "Assignements keywords",
+    "keyword.other"                is KeywordOther              -> [Keyword]                                         : "Other keywords, without a clear category",
+    "keyword.storage"              is KeywordStorage            -> [Keyword]                                         : "Default style for storage keywords",
+    "keyword.storage.modifier"     is KeywordStorageModifier    -> [KeywordStorage, Keyword]                         : "Storage modifier keywords, e.g. `move`, `dyn`",
+    "keyword.storage.modifier.mut" is KeywordStorageModifierMut -> [KeywordStorageModifier, KeywordStorage, Keyword] : "Mutable storage modifier",
+    "keyword.storage.modifier.ref" is KeywordStorageModifierRef -> [KeywordStorageModifier, KeywordStorage, Keyword] : "Reference storage modifier",
```

Comment:
Could the macro be modified so that you only need to define the first fallback, and it will recursively take the fallback of the fallback and so on?
That way you could just say something like:
```rust
    "keyword.operator"             is KeywordOperator           -> Keyword                 : "Operator keywords, e.g. `not`, `or`, etc.",
    "keyword.operator.assignment"  is KeywordOperatorAssignment -> KeywordOperator         : "Assignements keywords",
    "keyword.other"                is KeywordOther              -> Keyword                 : "Other keywords, without a clear category",
    "keyword.storage"              is KeywordStorage            -> Keyword                 : "Default style for storage keywords",
    "keyword.storage.modifier"     is KeywordStorageModifier    -> KeywordStorage          : "Storage modifier keywords, e.g. `move`, `dyn`",
    "keyword.storage.modifier.mut" is KeywordStorageModifierMut -> KeywordStorageModifier  : "Mutable storage modifier",
    "keyword.storage.modifier.ref" is KeywordStorageModifierRef -> KeywordStorageModifier  : "Reference storage modifier",
```
BTW, big fan of the idea of making the theme keys strongly typed :+1:

Reply:
I could do so with a custom iterator or using [`std::iter::successors`](https://doc.rust-lang.org/std/iter/fn.successors.html) yes, seems like a good idea and avoids a `SmallVec` too. Will do this evening